model/src/main/kotlin/PackageCurationData.kt

```diff
@@ -108,6 +109,14 @@ data class PackageCurationData(
     @JsonInclude(JsonInclude.Include.NON_NULL)
     val sourceCodeOrigins: List<SourceCodeOrigin>? = null
 ) {
+    init {
+        declaredLicenseMapping.values.forEach { spdxExpression ->
+            require(spdxExpression.isValid(ALLOW_LICENSEREF_EXCEPTIONS)) {
+                "The value '$spdxExpression' within the declared license mapping is not a valid SPDX expression."
```

Comment:
IMO "within" sounds a bit strange here. How about:

"The declared license mapping for '$key' maps to '$value' which is not a valid SPDX expression."

(So I would also include the key into the exception message for clarity.)